tests/Aspire.Hosting.Tests/WithUrlsTests.cs

```diff
@@ -396,50 +397,164 @@ public async Task MultipleUrlsForSingleEndpointAreIncludedInUrlSnapshot()
     }
 
     [Fact]
-    public async Task NonEndpointUrlsAreInactiveUntilResourceRunning()
+    public async Task UrlsAreInExpectedStateForResourcesGivenTheirLifecycle()
```

Comment:
Was hard to figure out what this test is testing since it looks like you are simulating what should happen, and then asserting that the simulation is correct vs internal behavior (might just be that there is a lot going on in this test).

Reply:
Yeah this test is verifying that the state of the URLs for project resources and custom resources is as expected as those resources move through their lifecycles.